lsm-srlinux/ci/Jenkinsfile

```
@@ -0,0 +1,170 @@
+        stage('Cleanup') {
+            steps {
+                withCredentials([
+                    file(credentialsId: 'JENKINS_JWE', variable: 'ENTITLEMENTS_FILE'),
+                    file(credentialsId: 'JENKINS_LICENSE', variable: 'LICENSE_KEY_FILE'),
+                ]) {
+                    sh '''
+                        cd "${WORKSPACE}/lsm-srlinux/containerlab"
+                        # The `clab destroy` command expects the com.inmanta.jwe and
+                        # com.inmanta.license files to exist in disk.
+                        mkdir resources
+                        cp "${ENTITLEMENTS_FILE}" resources/com.inmanta.jwe
+                        cp "${LICENSE_KEY_FILE}" resources/com.inmanta.license
+                        # The inmanta container is running as a different user then Jenkins
+                        chmod 444 resources/com.inmanta.jwe resources/com.inmanta.license
+                    '''
+                }
+                script {
+                    // Cleanup before starting
+                    sh(script: get_tear_down_script())
+                }
+            }
+        }
+        stage('Build') {
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                script {
+                    withCredentials([
+                        // Credentials to access the containers repository on Cloudsmith.
+                        usernamePassword(
+                            credentialsId: "cloudsmith-docker-token",
+                            usernameVariable: "CLOUDSMITH_DOCKER_USERNAME",
+                            passwordVariable: "CLOUDSMITH_DOCKER_TOKEN"
+                        ),
+                        // Credentials to access the inmanta-service-orchestrator-* repositories on Cloudsmith.
+                        string(
+                            credentialsId: "cloudsmith-token",
+                            variable: "CLOUDSMITH_PKGS_TOKEN",
+                        ),
+                    ]) {
+
+                        update_topology(env.version)
+                        update_project_yml(env.version, env.CLOUDSMITH_PKGS_TOKEN)
+                        sh '''
+                            cd "${WORKSPACE}/lsm-srlinux"
+                            echo "${CLOUDSMITH_DOCKER_TOKEN}" | sudo docker login -u "${CLOUDSMITH_DOCKER_USERNAME}" --password-stdin containers.inmanta.com
+                            # Ensure we have the latest version of the srlinux image
+                            sudo docker pull ghcr.io/nokia/srlinux:latest
+
+                            # Create python environment
+                            python3 -m venv venv
+                            source "venv/bin/activate"
+                            pip install -U pip
+                            if [ -n "${version}" ]; then
+                                pip install --pre pytest-inmanta-extensions "inmanta-service-orchestrator==${version}"
```

```suggestion
                                pip install --pre inmanta-core[pytest-inmanta-extensions] "inmanta-service-orchestrator==${version}"
```
This should ensure we get the matching `pytest-inmanta-extensions` version without having to backtrack too much.